src/main/java/com/gizasystems/automationexercise/apis/ApisProducts.java

```diff
@@ -10,7 +14,27 @@ public ApisProducts(SHAFT.API api) {
     }
 
     // Services
+    private static final String getAllProductsList_serviceName = "/productsList";
 
     //////////////////// Actions \\\\\\\\\\\\\\\\\\\\
+    @Step("API Get All Products List")
+    public ApisProducts getAllProductList() {
+        api.get(getAllProductsList_serviceName)
+                .setContentType(ContentType.URLENC)
+                .setTargetStatusCode(Apis.SUCCESS)
+                .perform();
+
+        return this;
+    }
+
+    //////////////////// Validations \\\\\\\\\\\\\\\\\\\\
+    @Step("Validate All products are listed")
+    public ApisProducts validateAllProductsAreListed(String expectedResponseFilePath) {
+        Response actualResponse = api.getResponse();
+        Validations.assertThat().response(actualResponse).matchesSchema(expectedResponseFilePath)
+                .perform();
```

Comment:
This is not how we use assertion with APIs with SHAFT now, please use `api.assertThatResponse()` method.
Also, You are only asserting one object not all the list; I think we need to assert on all the items not only one.